### PR TITLE
Update dependencies; replace unmaintained

### LINKED ISSUE #1521: RUSTSEC-2020-0053: dirs: dirs is unmaintained, use dirs-next instead
When `cargo-audit` is run on any crate that depends directly or indirectly on bat, the following is output (for more info, see https://rustsec.org/advisories/RUSTSEC-2020-0053):

```
Crate:         dirs
Version:       3.0.1
Warning:       unmaintained
Title:         dirs is unmaintained, use dirs-next instead
Date:          2020-10-16
ID:            RUSTSEC-2020-0053
URL:           https://rustsec.org/advisories/RUSTSEC-2020-0053
Dependency tree:
dirs 3.0.1
└── bat 0.17.1
```

It can easily be remedied by replacing dirs with dirs-next.

I would we willing to attempt to implement this and submit a PR if it would be accepted.

### PR DESCRIPTION
**Pull request by brightly-salty**
Fixes #1521 

Updates all dependencies
Replaces dirs with dirs-next
Replaces tempdir with tempfile

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -37,26 +34,26 @@ regex-fancy = ["syntect/regex-fancy"] # Use the rust-only "fancy-regex" engine
 
 [dependencies]
 atty = { version = "0.2.14", optional = true }
-ansi_term = "^0.12.1"
-ansi_colours = "^1.0"
+ansi_term = "0.12.1"
+ansi_colours = "1.0.1"
```
Comment (sharkdp) [inline Cargo.toml:38]: Why would we change this from `^1.0` to `1.0.1`?

Comment (sharkdp) [reply Cargo.toml:38]: Can we please revert these changes and only keep the `tempfile`/`dirs-next` ones?

Comment (brightly-salty) [reply Cargo.toml:38]: These were a side effect of `cargo update`. Let me manually revert

**Comment by sharkdp** [comment]
I manually downgraded `git2` to 0.13.15:
```
cargo update --precise 0.13.15 -p git2
```
as it doesn't support Rust 1.42, see https://github.com/rust-lang/git2-rs/pull/663#issuecomment-779640492

**Review by sharkdp — APPROVED** [review]

**Pull request merged**